### PR TITLE
No reductions are trivial

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/root_domain_map.cpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     // non-broadcast domains, except with a broadcast producer and a
     // consumer trivial reduction
     if (is_exact_ && producer_id->isBroadcast() != consumer_id->isBroadcast() &&
-        !(producer_id->isBroadcast() && consumer_id->isTrivialReduction())) {
+        !(producer_id->isBroadcast() && consumer_id->isReduction())) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/root_domain_map.cpp]: I think here we should just remove the condition of `!(producer-id->isBroadcast() && consumer_id->isReduction())`. This was to allow mappings of producer broadcasts and consumer trivial reductions, which shouldn't happen anymore.

**Review by naoyam — COMMENTED** [review]
Really nice cleanup! Just one comment on root domain mapping.

File: torch/csrc/jit/codegen/cuda/root_domain_map.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/root_domain_map.cpp]: Looks like I need to convert trivial reductions to squeeze in Welford before I can remove this. Will do it.

**Review by naoyam — APPROVED** [review]
LGTM. Great cleanup!

**Pull request merged**